### ISSUE TITLE
"Text" is too broad a name for geometry-only solution

### ISSUE THREAD

**Issue by jcarpenter** [ngokevin/aframe-text-component#1]
When I create 3D scenes and need text elements, I use textures more than 50% of the time. I expect that holds true for most people. We should integrate this geometry-based component with a mind to a future texture-based solution (hopefully the ability to render HTML textures directly in our scenes). 

That means either 1) choosing a more specific name for this component, or 2) integrating it as a valid primitive of the Geometry component.

**Comment by ngokevin** [comment]
That's valid. I can rename it to `text-geometry`.

**Comment by jcarpenter** [comment]
What do you think about the other option: adding it the valid attributes of the geometry component? That feels like a more logical fit to me.

**Comment by ngokevin** [comment]
It would work, but not sure about polluting and bloating that component. A separate text geometry component works just as well, and is more discoverable than hiding it within the geometry component. Makes documentation cleaner as well.

I think for sure, this will eventually have to be pulled in core from my playground. People are expecting it to be there.

**Comment by jcarpenter** [comment]
It's inconsistent, though, to define box, sphere, cone et al as being members of the geometry primitive, but text having it's own component (with a name that suggests geometry, no less). I see that as being confusing. 

Regarding discovery, I don't think component vs geometry moves the needle either way. The bulk of component discovery will be handled (as with everything else in A-Frame) through tutorials, examples, docs, future UIs, etc.

**Comment by ngokevin** [comment]
I like encouraging components to be modular. Anyone can write their own geometry-related components, like someone did for extrude/lathe. I'd encourage people to think of our basic geometry component as just a reference geometry component for basic shapes, not the catch-all component for every kind of geometry imaginable. 

Discovery can be impacted if someone scans a list of components on our docs and doesn't see any mention of a text component since it'd be tucked at the bottom of the geometry doc.

**Comment by jcarpenter** [comment]
I feel a text geometry creator falls into the "basic shapes" camp, whereas operations like Extrude, Lathe, Sweep, Boole etc are more complex "generators", requiring source paths and/or multiple objects. 

Prior art isn't terribly helpful in this case. In C4D and 3DS Max we build text objects by feeding a text spline into an extrude generator. But with your component the process is more simplified, requiring just a few parameters. Which makes it basically identical to other geometry primitives.

I'm particularly hung up on the distinction-without-difference of `geometry` vs `text-geometry`. That is ripe for unnecessary confusion. And I am not convinced of the downsides in the opposite direction.

**Comment by ngokevin** [comment]
I feel there's a lot of distinction. The parameters of the `text geometry` are of a different class than those of `geometry`: font size, font family, font weight.  And the typefonts have to be loaded in asynchronously.

And I actually feel it is more confusing it lump it into geometry. When people want text, they should just think "text component" rather than thinking "ok, it's a part of geometry" and dig through it. ```<a-entity geometry="primitive: text; text: HELLO; font: helvetica">``` doesn't feel like a good fit. I think "do one thing, do it well" should be applied here.

Maybe let's ask other people for more opinions on Slack.

**Comment by pinkwerks** [comment]
`text-geometry `seems better than `geometry="primitive=text...` as it avoids a taxonomy. I'd argue it keeps the component story simpler for the audience.

**Comment by ngokevin** [comment]
Yeah, once we move this to be a standard component, I'm thinking `text-geometry`.
